### DISCUSSION TITLE
Latest homebrew doctor said the /opt/homebrew is not writeable

### DISCUSSION THREAD

**Discussion by ngthphat** [Homebrew/discussions#3836] [Getting started] [answered]
I installed homebrew on my Mac M1 in a few months back, it works without any issues.

Today, I accidentally ran the homebrew install script. Then, whenever I instal whatever package, it always failed with this error

```
 phat.nguyen@Phats-MacBook-Pro  ~    brew tap homebrew/core
Error: /opt/homebrew is not writable. You should change the
ownership and permissions of /opt/homebrew back to your
user account:
  sudo chown -R $(whoami) /opt/homebrew
==> Tapping homebrew/core
fatal: could not create work tree dir '/opt/homebrew/Library/Taps/homebrew/homebrew-core': Permission denied
Error: Failure while executing; `git clone https://github.com/Homebrew/homebrew-core /opt/homebrew/Library/Taps/homebrew/homebrew-core --origin=origin --template=` exited with 128.
```

I tried to run the uninstall script, and re-install the homebrew, but the issue still there.

The installation happens smoothly without issue. However, when I run `brew doctor, this is what I got

```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: The following directories are not writable by your user:
/opt/homebrew
/opt/homebrew/etc/bash_completion.d
/opt/homebrew/share/doc
/opt/homebrew/share/man
/opt/homebrew/share/man/man1
/opt/homebrew/share/zsh
/opt/homebrew/share/zsh/site-functions
/opt/homebrew/var/homebrew/locks

You should change the ownership of these directories to your user.
  sudo chown -R $(whoami) /opt/homebrew /opt/homebrew/etc/bash_completion.d /opt/homebrew/share/doc /opt/homebrew/share/man /opt/homebrew/share/man/man1 /opt/homebrew/share/zsh /opt/homebrew/share/zsh/site-functions /opt/homebrew/var/homebrew/locks

And make sure that your user has write permission.
  chmod u+w /opt/homebrew /opt/homebrew/etc/bash_completion.d /opt/homebrew/share/doc /opt/homebrew/share/man /opt/homebrew/share/man/man1 /opt/homebrew/share/zsh /opt/homebrew/share/zsh/site-functions /opt/homebrew/var/homebrew/locks

Warning: Missing https://github.com/Homebrew/brew git origin remote.

Without a correctly configured origin, Homebrew won't update
properly. You can solve this by adding the remote:
  git -C "/opt/homebrew" remote add origin https://github.com/Homebrew/brew
==> Tapping homebrew/core
fatal: could not create work tree dir '/opt/homebrew/Library/Taps/homebrew/homebrew-core': Permission denied
Error: Failure while executing; `git clone https://github.com/Homebrew/homebrew-core /opt/homebrew/Library/Taps/homebrew/homebrew-core --origin=origin --template=` exited with 128.
Error: Failure while executing; `/opt/homebrew/bin/brew tap homebrew/core` exited with 1.
```

I followed the suggestion from the output above. But, nothing got changed.

**Comment by carlocab** [comment]
> I followed the suggestion from the output above. But, nothing got changed.

What was the output of following the suggestions from `brew doctor`?

**Comment by ngthphat** [comment, marked as answer]
@carlocab I found the root cause, there is an alias which mapping the `brew` command with another user. It explains why `brew` complains about permission. 
Thank you for your quick response.

This thread can be closed now.

**Reply by ainsliej** [reply]
I think I am having the same problem. Could you possibly elaborate on how you fixed the issue?

**Reply by cryptphil** [reply]
@ainsliej You can check if you have an alias for brew by running `alias | grep brew` and if it gives you a match just run `unalias brew
`.

**Reply by itfolksvfx** [reply]
Not for me
![image](https://github.com/user-attachments/assets/eb7a4b8b-08c0-4c9c-8131-3c501264e548)

**Comment by erlonbie** [comment]
I'm having the same problem, but I can't find any brew alias :/

**Reply by gromgit** [reply]
What's the output of the following:
```
type brew
brew config
brew doctor
```
?

**Reply by ricardocabral** [reply]
Same issue here:

```
brew is /opt/homebrew/bin/brew
HOMEBREW_VERSION: 4.3.19
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1f9bd2de89883ec9a59db3eab0f5475bce06d5bc
Last commit: 7 days ago
Core tap JSON: 06 Sep 21:06 UTC
Core cask tap JSON: 06 Sep 21:06 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_BAT_THEME: GitHub
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: code
HOMEBREW_MAKE_JOBS: 11
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: 11-core 64-bit arm_lobos
Clang: 15.0.0 build 1500
Git: 2.46.0 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: The following directories are not writable by your user:
/opt/homebrew

You should change the ownership of these directories to your user.
  sudo chown -R myuser /opt/homebrew

And make sure that your user has write permission.
  chmod u+w /opt/homebrew
```

**Reply by ricardocabral** [reply]
(and I have no alias for brew)

**Reply by ok-coder1** [reply]
@ricardocabral try running `sudo chown -R myuser /opt/homebrew` and `chmod u+w /opt/homebrew` as shown in the suggestions

**Reply by ricardocabral** [reply]
thanks @ok-coder1 sure, I've done that, and on the next time I try a command like `brew upgrade; brew cleanup -v -s --prune=all` the command fails and same suggestion is given.

**Reply by SMillerDev** [reply]
What is the owner of the directory?
`ls -al $(brew --prefix)` should give you that.

**Reply by SMillerDev** [reply]
Unrelated, but what are you using the additional `brew cleanup` command for? Because `brew upgrade` will also do a cleanup by default.

**Reply by ok-coder1** [reply]
> `sudo chown -R myuser /opt/homebrew` and `chmod u+w /opt/homebrew`



> thanks @ok-coder1 sure, I've done that, and on the next time I try a command like `brew upgrade; brew cleanup -v -s --prune=all` the command fails and same suggestion is given.

Did you change `myuser` to the name of your actual user? (For example, if your username is `xyz` then you would have to change the command accordingly like `sudo chown -R xyz /opt/homebrew`) 

> What is the owner of the directory? `ls -al $(brew --prefix)` should give you that.

Like @SMillerDev said, you can check if the directory is owned by you after running those commands.
